src/Spice86.Core/Emulator/VM/IPauseHandler.cs

```diff
@@ -0,0 +1,38 @@
+namespace Spice86.Core.Emulator.VM;
+
+/// <summary>
+/// Interface for handling pausing and resuming the emulator.
+/// </summary>
+public interface IPauseHandler {
+    /// <summary>
+    /// Gets a value indicating whether the emulator is currently paused.
+    /// </summary>
+    bool IsPaused { get; }
+
+    /// <summary>
+    /// Event triggered when a pause is requested on the emulator.
+    /// This allows other parts of the application to react to pauses,
+    /// such as stopping processing or updating UI elements.
+    /// </summary>
+    event EventHandler<EventArgs>? Pausing;
+
+    /// <summary>
+    /// Event triggered when a resume is requested on the emulator.
+    /// </summary>
+    event EventHandler<EventArgs>? Resumed;
```

Comment:
If the source object parameter isn't important, this could be an action<T> instead of an EventHandler<T>

EventHandler were made for code generators.

Reply:
Replaced with Action